src/dotdirdeps.cpp

```diff
@@ -13,55 +13,223 @@
 *
 */
 
+#include <algorithm>
+#include <iterator>
+#include <tuple>
 #include "dotdirdeps.h"
 
 #include "ftextstream.h"
 #include "util.h"
 #include "doxygen.h"
 #include "config.h"
 
+/** Properties are used to format the directories in the graph distinctively. */
+struct DotDirProperty
+{
+  bool isIncomplete = false; //!< true if not all successor of a cluster are drawn
+  bool isOrphaned = false; //!< true if parent is not drawn
+  bool isTruncated = false; //!< true has successors, none is drawn
+  bool isOriginal = false; //!< true if is the directory for which the graph is drawn
+  bool isPeriperal = false; //!< true if no successor of parent of original directory
+};
+
+/** Elements consist of (1) directory relation and (2) whether it is pointing only to inherited dependees. */
+typedef std::vector<std::tuple<const DirRelation*, bool>> DirRelations;
+typedef decltype(std::declval<DirDef>().level()) DirectoryLevel; //!< Integer for directory level.
+
+/** Returns a DOT color name according to the directory depth. */
+static QCString getDirectoryBackgroundColor(const DirectoryLevel depthIndex)
+{
+  return "/pastel19/" + QCString().setNum(depthIndex % 9 + 1);
+}
```

Comment:
The reason for this solution is, that it was easy to implement and it can cope with any number of levels without requiring much information. A [Brewer colour scheme](http://www.graphviz.org/doc/info/colors.html#brewer) is applied.

Colour may also be a subject of taste. I am open for other suggestions.